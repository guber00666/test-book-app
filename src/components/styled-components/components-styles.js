import styled from "styled-components";

export const ListStyles = styled.div `
              margin-top: 30px;
              border: solid 1px;
              border-radius: 0.3em;
        `;
export const PaginatorStyles = styled.div `
              margin-top: 30px;
              margin-bottom: 30px;
              float: right;
        `;
export const AppStyles = styled.div `
          margin-top: 30px;
    `;
export const CriteriaStyles = styled.div `
             margin-top: 30px;
             font-weight: bold;
    `;
export const FormStyles = styled.div`
             border: solid 1px;
             border-radius: 0.3em;
    `;
export const SearchPanelStyles = styled.div `
    border: solid 1px;
    border-radius: 0.3em;
`;