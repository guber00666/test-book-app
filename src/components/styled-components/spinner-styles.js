import styled, {keyframes} from "styled-components";

export const SpinnerPulse  = styled.div `
                width: 200px;
                height: 200px;
                display: inline-block;
                overflow: hidden;
                background: #ffffff;
                margin-left: 26em;
        `;
const SpinnerElements1 = keyframes`
    0% { top: 39.199999999999996px; height: 121.60000000000001px }
    50% { top: 62px; height: 76px }
    100% { top: 62px; height: 76px }
`;
const SpinnerElements2 = keyframes`
    0% { top: 44.89999999999999px; height: 110.20000000000002px }
    50% { top: 62px; height: 76px }
    100% { top: 62px; height: 76px }
`;
const SpinnerElements3 = keyframes`
    0% { top: 50.6px; height: 98.8px }
    50% { top: 62px; height: 76px }
    100% { top: 62px; height: 76px }
`;
export const PulseElement = styled.div`
        width: 100%;
        height: 100%;
        position: relative;
        transform: translateZ(0) scale(1);
        backface-visibility: hidden;
        transform-origin: 0 0;
    `;
export const NthChild1 = styled.div `
         position: absolute; 
         width: 28px;
         box-sizing: content-box;
         left: 36px;
         background: #353535;
         animation: ${SpinnerElements1} 1s cubic-bezier(0,0.5,0.5,1) infinite;
         animation-delay: -0.2s
    `;
export const NthChild2 = styled.div `
          position: absolute; 
          width: 28px;
          box-sizing: content-box;
          left: 86px;
          background: #666666;
          animation: ${SpinnerElements2} 1s cubic-bezier(0,0.5,0.5,1) infinite;
          animation-delay: -0.1s
    `;
export const NthChild3 = styled.div `
           position: absolute; 
           width: 28px;
           box-sizing: content-box;
           left: 136px;
           background: #9b9b9b;
           animation: ${SpinnerElements3} 1s cubic-bezier(0,0.5,0.5,1) infinite;
           animation-delay: initial;
    `;