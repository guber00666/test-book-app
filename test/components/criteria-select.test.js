import React from "react";
import Adapter from "enzyme-adapter-react-16/build";
import CriteriaSelect from '../../src/components/criteria-select';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { CriteriaStyles, FormStyles } from "../../src/components/styled-components/components-styles";
import configureStore from 'redux-mock-store'
import 'jest-styled-components';
import {
    mapStateToCriteriaProps,
    mapDispatchToCriteriaProps
} from '../../src/actions/index';

configure({ adapter: new Adapter() });

const initialState = {
    criteriaValue: "0"
};

const middlewares = [];
const mockStore = configureStore(middlewares);
const store = mockStore(initialState);

describe("<CriteriaSelect />",() => {

    const wrapperCriteria = mount(
        <Provider store={store}>
            <CriteriaSelect />
        </Provider>
    );

    it('CriteriaSelect should render self and other imports', () => {
        expect(wrapperCriteria.find('section').hasClass("criteria-select")).toBe(true);
        expect(wrapperCriteria.find('div').at(1).hasClass("form-group")).toBe(true);
        expect(wrapperCriteria.find('label').text()).toBe('Criteria select');
        expect(wrapperCriteria.find('select').hasClass("form-control")).toBe(true);

        expect(wrapperCriteria.find('CriteriaStyles')).toBeDefined();
        expect(wrapperCriteria.find('FormStyles')).toBeDefined();

        expect(wrapperCriteria).toMatchSnapshot();
    });

    it('CriteriaSelect should have styles', () => {
        const criteriaStyles = renderer.create(<CriteriaStyles />).toJSON();
        expect(criteriaStyles).toHaveStyleRule('margin-top', '30px');
        expect(criteriaStyles).toHaveStyleRule('font-weight', 'bold');

        const formStyles = renderer.create(<FormStyles />).toJSON();
        expect(formStyles).toHaveStyleRule('border', 'solid 1px');
        expect(formStyles).toHaveStyleRule('border-radius', '0.3em');

        expect(criteriaStyles).toMatchSnapshot();
        expect(formStyles).toMatchSnapshot();
    });

    it('check select correct working', () => {
       const event = {target: {value: '1'}};
       expect(wrapperCriteria.find('option').at(0).prop('value')).toEqual('0');
       wrapperCriteria.find('option').at(1).simulate('change', event);
       wrapperCriteria.update();
       expect(wrapperCriteria.find('option').at(1).prop('value')).toEqual('1');
    });

    it('criteriaValue should show previously rolled value', () => {
        expect(mapStateToCriteriaProps(initialState).criteriaValue).toEqual("0");
    });

    it('criteriaValue should roll the dice again when onChange work out', () => {
        const dispatch = jest.fn();

        mapDispatchToCriteriaProps(dispatch).setCriteriaValue();
        expect(dispatch.mock.calls[0][0]).toEqual({ type: 'SET_CRITERIA' });
    });
});