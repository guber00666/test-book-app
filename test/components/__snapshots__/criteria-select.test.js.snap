// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CriteriaSelect /> CriteriaSelect should have styles 1`] = `
.c0 {
  margin-top: 30px;
  font-weight: bold;
}

<div
  className="c0"
/>
`;

exports[`<CriteriaSelect /> CriteriaSelect should have styles 2`] = `
.c0 {
  border: solid 1px;
  border-radius: 0.3em;
}

<div
  className="c0"
/>
`;

exports[`<CriteriaSelect /> CriteriaSelect should render self and other imports 1`] = `
.c0 {
  margin-top: 30px;
  font-weight: bold;
}

.c1 {
  border: solid 1px;
  border-radius: 0.3em;
}

<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CriteriaSelect)>
    <CriteriaSelect
      criteriaValue="0"
      setCriteriaValue={[Function]}
    >
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": false,
                "lastClassName": "c0",
                "rules": Array [
                  "
             margin-top: 30px;
             font-weight: bold;
    ",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c0"
          >
            <section
              className="criteria-select"
            >
              <div
                className="form-group"
              >
                <label>
                  Criteria select
                </label>
                <styled.div>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-ifAKCX",
                          "isStatic": false,
                          "lastClassName": "c1",
                          "rules": Array [
                            "
             border: solid 1px;
             border-radius: 0.3em;
    ",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-ifAKCX",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="c1"
                    >
                      <select
                        className="form-control"
                        onChange={[Function]}
                      >
                        <option
                          key="0"
                          value="0"
                        >
                          Book Name
                        </option>
                        <option
                          key="1"
                          value="1"
                        >
                          Book Genre
                        </option>
                        <option
                          key="2"
                          value="2"
                        >
                          Book Author
                        </option>
                      </select>
                    </div>
                  </StyledComponent>
                </styled.div>
              </div>
            </section>
          </div>
        </StyledComponent>
      </styled.div>
    </CriteriaSelect>
  </Connect(CriteriaSelect)>
</Provider>
`;
